### PR TITLE
API for returning all rooms with its players

### PR DESCRIPTION
**Pull request by stathismor**
Return Rooms with they players. Later on other info will be relevant, probably state related.

image

### CODE REVIEW

File: packages/server/src/Server.ts [hunk 1]
```diff
@@ -10,8 +11,9 @@ export const createServer = () => {
   app.use(cors({ origin: process.env.HTTP_ORIGIN }));
 
   // Application routing
-  app.get('/users', (req: Request, res: Response, next: NextFunction) => {
-    res.status(200).send({ data: ['user1', 'user2'] });
+  app.get('/rooms', async (req: Request, res: Response, next: NextFunction) => {
```
Comment (dbertella) [inline packages/server/src/Server.ts:14]: Why do you think an endpoint is better in this case? I expect this info to change quite a lot with players adding new rooms so I think it should still be a socket event. Unless we do rest first and update later.

File: packages/server/src/database/redisClient.ts [hunk 2]
```diff
@@ -5,13 +5,43 @@ import * as util from 'util';
...
 const setAsync = util.promisify(client.set).bind(client);
+const mgetAsync: (args: string[]) => Promise<string[]> = util.promisify(client.mget).bind(client);
+const scanAsync: (cursor: string, ...args: string[]) => Promise<[string, string[]]> = util
+  .promisify(client.scan)
+  .bind(client);
+const scanAll = async (pattern: string) => {
+  const found: string[] = [];
+  let cursor = '0';
+
+  do {
```
Comment (dbertella) [inline packages/server/src/database/redisClient.ts:16]: Do while lol it's something I didn't think I would have seen in a codebase. Is this how you get data from redis? Why do we need to check elements by their name?

File: packages/server/src/Server.ts (hunk 1 above)
Comment (stathismor) [reply packages/server/src/Server.ts:14]: Yeah, you are right, will move this to socket.

File: packages/client/src/components/RoomTable.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    getRooms.then((data) => setRooms(data));
+  }, []);
+
+  return (
+    <Box>
+      <Heading as="h2">Rooms</Heading>
+      <Box as="ul">
+        {rooms.map((room) => (
+          <Flex key={room.name} as="li" bg="primary" sx={{ flexDirection: 'column', mt: 1 }}>
+            <Flex>
```
Comment (dbertella) [inline packages/client/src/components/RoomTable.tsx]: you don't need this flex

File: packages/client/src/components/RoomTable.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+              Room:<strong>{room.name}</strong>
```
Comment (dbertella) [inline packages/client/src/components/RoomTable.tsx]: I think you want to add a space here

File: packages/client/src/components/RoomTable.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  useEffect(() => {
+    getRooms.then((data) => setRooms(data));
+  }, []);
+
+  return (
+    <Box>
+      <Heading as="h2">Rooms</Heading>
+      <Box as="ul">
+        {rooms.map((room) => (
+          <Flex key={room.name} as="li" bg="primary" sx={{ flexDirection: 'column', mt: 1 }}>
```
Comment (dbertella) [inline packages/client/src/components/RoomTable.tsx]: use Card component from theme-ui

**Pull request merged**